src/SAML2/XML/saml/AuthnContext.php

```diff
@@ -0,0 +1,266 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SAML2\XML\saml;
+
+use DOMElement;
+use SAML2\Constants;
+use SAML2\DOMDocumentFactory;
+use SAML2\Utils;
+use SAML2\XML\saml\AuthenticatingAuthority;
+use SAML2\XML\saml\AuthnContextClassRef;
+use SAML2\XML\saml\AuthnContextDeclRef;
+use SAML2\XML\saml\AuthnContextRef;
+use Webmozart\Assert\Assert;
+
+/**
+ * Class representing SAML2 AuthnContext
+ *
+ * @author Tim van Dijen, <tvdijen@gmail.com>
+ * @package SimpleSAMLphp
+ */
+class AuthnContext extends \SAML2\XML\AbstractConvertable
+{
+    /** @var \SAML2\XML\saml\AuthnContextClassRef|null */
+    private $authnContextClassRef = null;
+
+    /** @var \SAML2\XML\saml\AuthnContextDeclRef|null */
+    private $authnContextDeclRef = null;
+
+    /** @var \SAML2\XML\saml\AuthnContextDecl|null */
+    private $authnContextDecl = null;
+
+    /** @var \SAML2\XML\saml\AuthenticatingAuthority[]|null */
+    private $authenticatingAuthorities = null;
+
+
+    /**
+     * Initialize an AuthnContext.
+     *
+     * @param \SAML2\XML\saml\AuthnContextClassRef|null $authnContextClassRef
+     * @param \SAML2\XML\saml\AuthnContextDecl|null $authnContextDecl
+     * @param \SAML2\XML\saml\AuthnContextDeclRef|null $authnContextDeclRef
+     * @param \SAML2\XML\saml\AuthenticatingAuthority[]|null $authenticatingAuthorities
+     */
+    public function __construct(
+        ?AuthnContextClassRef $authnContextClassRef,
+        ?AuthnContextDecl $authnContextDecl,
+        ?AuthnContextDeclRef $authnContextDeclRef,
+        ?array $authenticatingAuthorities
+    ) {
+        if (!is_null($authnContextClassRef)) {
+            Assert::oneOf(
+                null,
+                [$authnContextDecl, $authnContextDeclRef],
+                'Can only have one of AuthnContextDecl/AuthnContextDeclRef'
+            );
+        } else {
+            Assert::false(
+                is_null($authnContextDecl) && is_null($authnContextDeclRef),
+                'You need either an AuthnContextDecl or an AuthnContextDeclRef'
+            );
+        }
+
+        $this->setAuthnContextClassRef($authnContextClassRef);
+
+        if (!is_null($authnContextDecl)) {
```

Comment:
Given that the setters have nullable parameters, we can skip these `is_null()` checks and just call the setters.